### PR TITLE
Clean up mirror and support quay.io pushing

### PR DESCRIPTION
**Pull request by smarterclayton**
quay.io doesn't have schema2 support yet, tolerate it in oc image mirror. Also add file support.

Does some preparator refactoring for adding parallel pushing support.

### CODE REVIEW

**Comment by dmage** [comment]
@smarterclayton is it a typo or does quay really not have schema *1* support?

**Comment by smarterclayton** [comment]
Typo, they lack schema2 (are working on it now)

**Comment by smarterclayton** [comment]
Any comments?  I'm trying to get mirror ready to sync lots of images to quay, then I'll start adding test suites around it.  More iterative dev right now so we can get to a beta state for this command.

File: pkg/oc/cli/cmd/image/mirror/mirror.go [hunk 1]
```diff
@@ -162,56 +173,109 @@ func parseDestination(ref string) (imageapi.DockerImageReference, DestinationTyp
...
+	for s.Scan() {
+		line := s.Text()
+
+		// remove comments and whitespace
+		if i := strings.Index(line, "#"); i != -1 {
+			line = line[0:i]
+		}
+		line = strings.TrimSpace(line)
+		if len(line) == 0 {
+			continue
```
Comment (dmage) [inline pkg/oc/cli/cmd/image/mirror/mirror.go]: It will go to the next line without incrementing lineNumber.

**Comment by dmage** [comment]
Converting schema 2 into schema 1 looks good. Everything else doesn't seem changed.

File: pkg/oc/cli/cmd/image/mirror/mirror.go (hunk 1 above)
Comment (smarterclayton) [reply pkg/oc/cli/cmd/image/mirror/mirror.go]: thanks

**Comment by smarterclayton** [comment]
Thanks, updated

**Comment by smarterclayton** [comment]
Tagging based on comments, let me know if there's something still to change and I'll do it in the follow on to this PR (adding planning for a mirror)

**Comment by smarterclayton** [comment]
stupid job, work

**Pull request merged**